### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by berylcl**
In this project i was able to 
:large_blue_diamond:Remove all hardcoded items from the tasks array.
:large_blue_diamond:Create a new JavaScript file for the new functionality.
:large_blue_diamond:Implement a function for adding a new task (add a new element to the array).
:large_blue_diamond:Implement a function for deleting a task (remove an element from the array)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,28 +1,117 @@
...
-  { description: 'Wash utensils', completed: true, index: 3 },
-];
+let tasks = [];
+if (localStorage.getItem('tasks')) {
+  tasks = JSON.parse(localStorage.getItem('tasks'));
+}
+function saveTasks() {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+function addTask(description) {
```
Comment (wayungi) [inline src/index.js:15]: - [ OPTIONAL] I suggest you consider using arrow functions for all your functions whenever possible. They are concise and easy to read and have an implicit return. 🤔

File: src/index.js [hunk 2]
```diff
@@ -1,28 +1,117 @@
...
+      input.addEventListener('keydown', (e) => {
+        if (e.key === 'Enter') {
+          const newDesc = input.value.trim();
+          if (newDesc.length > 0) {
+            task.description = newDesc;
+            taskDesc.innerText = newDesc;
+            input.replaceWith(taskDesc);
+            saveTasks();
+          }
+        }
```
Comment (wayungi) [inline src/index.js:76]: - [ OPTIONAL] Good work implementing the edit feature. I suggest you make the input field for the edit action consistent with the display tasks
edit

File: src/index.js [hunk 3]
```diff
@@ -1,28 +1,117 @@
...
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+function addTask(description) {
+  const task = {
+    description,
+    completed: false,
+    index: tasks.length,
+  };
+  tasks.push(task);
+  saveTasks();
```
Comment (wayungi) [inline src/index.js]: - [ ] Please note that new tasks should have their index set to the value of the length of the array. This means the index of the elements should start from 1, not zero. What do you think 🤔 ?
index

**Review by wayungi — CHANGES_REQUESTED** [review]
Hi @berylcl , 

image


**Congratulations on the good job done so far!** 🤝 

There are some issues that you still need to work on to go to the next project but you are almost there! 😄 

## Highlights ⚡ ⚡ 
- [ ] All linter checks are passing ✔️ 
- [ ]  Correct GitHub flow ✔️ 
- [ ] Descriptive PR body ✔️ 

## Required Changes ♻️
_Check the comments under the review._
- [ ] Your **live demo link** is broken. If this project does not require deployment kindly  delete this section or add a live demo link that is functional 😄 
live

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @berylcl 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git -Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.


## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**